--- a/modules/invenio-oauth2server/invenio_oauth2server/validators.py
+++ b/modules/invenio-oauth2server/invenio_oauth2server/validators.py
@@ -10,6 +10,7 @@

 from __future__ import absolute_import, print_function

+import os
 from flask import current_app
 from oauthlib.oauth2.rfc6749.errors import InsecureTransportError, \
     InvalidRedirectURIError
@@ -32,6 +33,8 @@ def validate_redirect_uri(value):
     if sch != 'https':
         if ':' in netloc:
             netloc, port = netloc.split(':', 1)
+        if os.environ.get('OAUTHLIB_INSECURE_TRANSPORT'):
+            return
         if not (netloc in ('localhost', '127.0.0.1') and sch == 'http'):
             raise InsecureTransportError()

